File: packages/ogdf.js/src/entry/energybased/SpringEmbedderKK.cpp
#include <ogdf/energybased/SpringEmbedderKK.h>
#include "../main.h"

EM_PORT_API(SpringEmbedderKK *)
LayoutModule_SpringEmbedderKK(bool computeMaxIterations, double desLength, int globalIterationFactor, int maxGlobalIterations, int maxLocalIterations, double stopTolerance, bool useLayout, double zeroLength)
{
	//LayoutModule
	SpringEmbedderKK *model = new SpringEmbedderKK();

	model->computeMaxIterations(computeMaxIterations);
	model->setDesLength(desLength);
	model->setGlobalIterationFactor(globalIterationFactor);
	model->setMaxGlobalIterations(maxGlobalIterations);
	model->setMaxLocalIterations(maxLocalIterations);
	model->setStopTolerance(stopTolerance);
	model->setUseLayout(useLayout);
	model->setZeroLength(zeroLength);

	return model;
}

File: packages/ogdf.js/src/layouts/energybased/SpringEmbedderKK.js
import { PARAMETER_TYPE } from '../../utils/parameters'

export default {
    computeMaxIterations: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: true
    },
    desLength: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [-Infinity, Infinity],
        default: 0.0
    },
    globalIterationFactor: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 16
    },
    maxGlobalIterations: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 2147483647
    },
    maxLocalIterations: {
        type: PARAMETER_TYPE.INT,
        range: [0, Infinity],
        default: 2147483647
    },
    stopTolerance: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [0, Infinity],
        default: 0.001
    },
    useLayout: {
        type: PARAMETER_TYPE.BOOL,
        range: [true, false],
        default: true
    },
    zeroLength: {
        type: PARAMETER_TYPE.DOUBLE,
        range: [-Infinity, Infinity],
        default: -1.0
    }
}
